### ISSUE TITLE
geo is not searchable

### ISSUE THREAD

**Issue by lorenzorietti** [bionode/bionode-ncbi#20]
add mapping to gst

**Comment by Istar-Eldritch** [comment]
As I see it we have some options:

- **Have an internal list of the valid db options and check it upfront request**
    - This can be created the first time the users uses the tool and use a cached version of it from that moment if hardcoding is not an option.
    - This will also allow the usage of a [string metric algorithm](https://en.wikipedia.org/wiki/String_metric) for suggestions in case the db is not found.
- **Identify conflictive cases and create the specific mapping for them.**
    - This is a simple and fast solution. But won't protect the user from misspells.
-  **Do the request but if the following object is  returned bail out and notify the user:**

  ```json
  {
    "header": {
      "type": "esearch",
      "version": "0.3"
    },
    "esearchresult": {
      "ERROR": "Invalid db name specified: dst"
    }
  }
  ```
  - The check can be done here [bionode-ncbi.js#L523](https://github.com/bionode/bionode-ncbi/blob/master/lib/bionode-ncbi.js#L523) skipping the retry loop.
  - An internal list of valid dbs can be present for suggestions as in the first case

In my opinion the last option is the most suitable. @bmpvieira If you agree I'll create a pull request to solve it. Do you have any other idea?

**Comment by bmpvieira** [comment]
I think we should go with the first option, check if db is valid and if not present list of valid ones (hardcoded in an internal config file in JSON or YAML format).

Here's one retrieved from the select box on [NCBI website](https://www.ncbi.nlm.nih.gov) (not sure if all work with eutils/biondo-ncbi):
```csv
"gquery","All Databases"
"assembly","Assembly"
"bioproject","BioProject"
"biosample","BioSample"
"biosystems","BioSystems"
"books","Books"
"clinvar","ClinVar"
"clone","Clone"
"cdd","Conserved Domains"
"gap","dbGaP"
"dbvar","dbVar"
"nucest","EST"
"gene","Gene"
"genome","Genome"
"gds","GEO DataSets"
"geoprofiles","GEO Profiles"
"nucgss","GSS"
"gtr","GTR"
"homologene","HomoloGene"
"medgen","MedGen"
"mesh,"MeSH"
"ncbisearch","NCBI Web Site"
"nlmcatalog","NLM Catalog"
"nuccore","Nucleotide"
"omim","OMIM"
"pmc","PMC"
"popset","PopSet"
"probe","Probe"
"protein","Protein"
"proteinclusters","Protein Clusters"
"pcassay","PubChem BioAssay"
"pccompound","PubChem Compound"
"pcsubstance","PubChem Substance
"pubmed,"PubMed"
"pubmedhealth","PubMed Health"
"snp","SNP"
"sparcle","Sparcle"
"sra","SRA"
"structure","Structure"
"taxonomy","Taxonomy"
"toolkit","ToolKit"
"toolkitall","ToolKitAll"
"toolkitbook","ToolKitBook"
"toolkitbookgh","ToolKitBookgh"
"unigene","UniGene"
```

**Comment by Istar-Eldritch** [comment]
I'll put some time today into this.

**Comment by bmpvieira** [comment]
I've kind of solved this at the CLI level by switching to yargs (https://github.com/bionode/bionode-ncbi/pull/32) but I think it would be nice to also have these kind of checks internally in the library. Maybe we could somehow reuse yargs `choice` definition.

**Comment by bmpvieira** [comment]
Fixed by #33.  Thanks!
